packages/example/src/components/AddSnap/AddSnap.tsx

```diff
@@ -0,0 +1,36 @@
+import { Button, Result, Spin } from 'antd';
+import { FC, useEffect, useState } from 'react';
+import { ISnap } from '../../hooks/useSnap';
+
+interface IAddSnap {
+  snap: ISnap;
+}
+
+export const AddSnap: FC<IAddSnap> = ({ snap }) => {
+  const [initialSpinner, setInitialSpinner] = useState(true);
+
+  //initial spinner for better UX while isMetaMaskFlaskAvailable function promise race finishes
+  useEffect(() => {
+    void (async () => {
+      return await new Promise<void>(() =>
+        setTimeout(() => {
+          setInitialSpinner(false);
+          return;
+        }, 1000),
+      );
+    })();
+  }, []);
```

Comment:
maybe better to move this logic inside a snap, and add states like `isReady` for this kind of spinner